src/black/__init__.py

```diff
@@ -626,8 +644,6 @@ def get_sources(
                     gitignore,
                 )
             )
-        elif s == "-":
-            sources.add(p)
         elif p.is_file():
```

Comment:
```suggestion
        elif p.is_file() or is_stdin:
```

With a custom stdin filepath given, Black will try to format it in place as if it was a file:

```console
(black) ichard26@acer-ubuntu:~/programming/oss/black$ echo "print('hello world')" | black -v --stdin-filename="hello.py" -
Using configuration from /home/ichard26/programming/oss/black/pyproject.toml.
Traceback (most recent call last):
  File "/home/ichard26/programming/oss/black/src/black/__init__.py", line 706, in reformat_one
    if changed is not Changed.CACHED and format_file_in_place(
  File "/home/ichard26/programming/oss/black/src/black/__init__.py", line 845, in format_file_in_place
    then = datetime.utcfromtimestamp(src.stat().st_mtime)
  File "/opt/python3.8.5/lib/python3.8/pathlib.py", line 1194, in stat
    return self._accessor.stat(self)
FileNotFoundError: [Errno 2] No such file or directory: 'hello.py'
error: cannot format hello.py: [Errno 2] No such file or directory: 'hello.py'
Oh no! 💥 💔 💥
1 file failed to reformat.
```

This is because `Path("-").is_file` returns `False` because a file doesn't actually exist at that path ([See `pathlib.Path.is_file` docs](https://docs.python.org/3.8/library/pathlib.html#pathlib.Path.is_file)). Therefore the path isn't prefixed with the magic string that tells `reformat_one` to use `format_stdin_to_stdout` and not `format_file_in_place`. Also `force-exclude` wouldn't be checked, but Black trying to format a file that doesn't exist is a bigger problem without this suggestion :-)